### ISSUE TITLE
Build a Docker container

### ISSUE THREAD

**Issue by ewels** [seqeralabs/tower-cli#136]
Seeing as Nextflow is all about containerisation, it would be nice to have this cli readily available in a container 😉 

Could build and host via the GitHub container registry etc. Or wherever really.

**Comment by jordeu** [comment]
Good idea. 

It is possible to use [jreleaser](https://jreleaser.org/guide/latest/configuration/packagers/docker.html) to publish it. 

Related #128

**Comment by ewels** [comment]
Yup, or just [GitHub actions](https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages)

The `Dockerfile` I just wrote for myself is pretty simple and seems to work well:

```dockerfile
FROM alpine

ARG TOWER_CLI_VERSION="0.4"

# Install Tower CLI
RUN apk add --no-cache curl ca-certificates
RUN curl -L https://github.com/seqeralabs/tower-cli/releases/download/v${TOWER_CLI_VERSION}/tw-${TOWER_CLI_VERSION}-linux-x86_64 > tw
RUN chmod +x ./tw
RUN mv tw /usr/local/bin/
```

**Comment by ewels** [comment]
But of course for automation here you don't want to be downloading releases etc, so yeah might make sense to use `jreleaser`. I'm just not familiar with it.

**Comment by drpatelh** [comment]
Now that the Tower CLI has been added to Bioconda, we just have to stay on top of bumping versions there and we will get Biocontainers for free:
https://quay.io/repository/biocontainers/tower-cli?tag=latest&tab=tags

x:ref: https://github.com/seqeralabs/tower-cli/issues/293

**Comment by ewels** [comment]
The biocontainers image is outside of our control, that's managed via the BioConda + BioContainers communities.

But we could definitely build our own docker images to use as an alternative, using whatever tags we want.

**Comment by ewels** [comment]
I think it would relatively easy to do. The releases are already going out via GitHub actions. We could use JReleaser to do this directly as suggested above.

* [config file](https://github.com/seqeralabs/tower-cli/blob/master/jreleaser.yml)
* [Updated JReleaser docs link](https://jreleaser.org/guide/latest/reference/packagers/docker.html) (link above is now broken)

Looks like JReleaser has support for pushing the images to GitHub Packages, which would probably be the easiest (`jreleaser.github.username`, `jreleaser.gitlab.username` etc).
